packages/core/contracts-ovm/insured-bridge/implementation/BridgePool.sol

```diff
@@ -211,15 +215,63 @@ contract BridgePool is Testable {
     /**
      * @notice Reward relayers if a pending relay price request has a price available on the OptimisticOracle. Mark
      * the relay as complete.
-     * @dev Caller must have approved this contract to spend the deposit amount of L1 tokens to relay. There can only
-     * be one instant relayer per relay attempt and disputed relays cannot be sped up.
-     * @param _depositData Unique set of L2 deposit data that caller is trying to instantly relay.
+     * @param _depositData Unique set of L2 deposit data that caller is trying to settle a relay for.
      */
     function settleRelay(DepositData memory _depositData) public {
         bytes32 depositHash = _getDepositHash(_depositData);
         RelayData storage relay = relays[depositHash];
         require(relays[depositHash].relayState == RelayState.Pending, "Can only speed up pending slow relay");
-        // TODO: Pay relayer rewards using PendingRelay data.
+        require(
+            _getOptimisticOracle().hasPrice(
+                address(this),
+                bridgeAdmin.identifier(),
+                relay.priceRequestTime,
+                getRelayAncillaryData(_depositData, relay)
+            ),
+            "OptimisticOracle has not resolved relay price request"
+        );
+
+        // Note: Why don't we have to check the value of the price?
+        // - If the OptimisticOracle has a price and the relayState is PENDING, then we can safely assume that the relay
+        // was validated. This is because this contract proposes a price of 1e18, or "YES" to the identifier posing the
+        // question "Is this relay valid for the associated deposit?". If the proposal is disputed, then the relayState
+        // will be reset to UNINITIALIZED. If the proposal is not disputed, and there is a price available, then the
+        // proposal must have passed the dispute period.
+
+        relay.relayState = RelayState.Finalized;
+
+        // Pay recipient the deposit amount less fees.
+        IERC20(_depositData.l1Token).safeTransfer(
+            _depositData.recipient,
+            _depositData.amount - _getRealizedFeeAmount(relay.realizedFeePct, _depositData.amount)
+        );
+
+        // Reward relayers.
+        // TODO: For now, if there was an instant relay associated with this deposit, then split the reward 50/50
+        // between slow and instant relayer. Otherwise pay the slow relayer the full reward.
+        uint256 rewardPool = _getProposerRewardAmount(relay.proposerRewardPct, _depositData.amount);
+        if (relay.instantRelayer != address(0)) {
```

Comment:
this bit of branching logic is really important as it defines the kind of payout done. please add another comment on this.